Script/update-versions.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+# make this script executable from terminal:
+# chmod 755 update-versions.sh
+
+set -e # Any subsequent(*) commands which fail will cause the shell script to exit immediately
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+LINE="================================================================================"
+VERSION_REGEX="[0-9]+\.[0-9]+\.[0-9]+"
+DEPENDENCIES=none
+
+# make a "dictionary" to help us find the correct spm repo per dependency (if necessary)
+# IMPORTANT - this will be used in a regex search so escape special chars
+# usage :
+# getRepo AEPCore
+
+declare "repos_AEPCore=https:\/\/github\.com\/adobe\/aepsdk-core-ios\.git"
+getRepo() {
+    local extensionName=$1
+    local url="repos_$extensionName"
+    echo "${!url}"
+}
+
+help()
+{
+   echo ""
+   echo "Usage: $0 -n EXTENSION_NAME -v NEW_VERSION -d \"PODSPEC_DEPENDENCY_1, PODSPEC_DEPENDENCY_2\""
+   echo ""
+   echo -e "    -n\t- Name of the extension getting a version update. \n\t  Example: Edge, Analytics\n"
+   echo -e "    -v\t- New version to use for the extension. \n\t  Example: 3.0.2\n"
+   echo -e "    -d (optional)\t- Dependency(ies) that require updating in the extension's podspec and Package.swift file. \n\t  Example: -d \"AEPCore 3.7.3\" (update the dependency on AEPCore to version 3.7.3 or newer)\n"
+   exit 1 # Exit script after printing help
+}
+
+while getopts "n:v:d:" opt
+do
+   case "$opt" in
+      n ) NAME="$OPTARG" ;;
+      v ) NEW_VERSION="$OPTARG" ;;
+      d ) DEPENDENCIES="$OPTARG" ;;      
+      ? ) help ;; # Print help in case parameter is non-existent
+   esac
+done
+
+# Print help in case parameters are empty
+if [ -z "$NAME" ] || [ -z "$NEW_VERSION" ]
+then
+   echo "********** USAGE ERROR **********"
+   echo "Some or all of the parameters are empty.  See usage below:";
+   help
```

Comment:
you might want to exit early for missing required params when running this in a non-interactive mode like the gh action